ipams/remote/api/api.go

```diff
@@ -0,0 +1,81 @@
+// Package api defines the data structure to be used in the request/response
+// messages between libnetwork and the remote ipam plugin
+package api
+
+import (
+	"net"
+)
+
+// Response is the basic response structure used in all responses
+type Response struct {
+	Error string
+}
+
+// IsSuccess returns wheter the plugin response is successful
+func (r *Response) IsSuccess() bool {
+	return r.Error == ""
+}
+
+// GetError returns the error from the response, if any.
+func (r *Response) GetError() string {
+	return r.Error
+}
+
+// GetAddressSpacesResponse is the response to the ``get default address spaces`` request message
+type GetAddressSpacesResponse struct {
+	Response
+	LocalDefaultAddressSpace  string
+	GlobalDefaultAddressSpace string
+}
+
+// RequestPool represents the expected data in a ``request address pool`` request message
+type RequestPool struct {
```

Comment:
To be consistent with the remote network driver and the REST API, these type should be call `<API>Request` and `<API>Response`.   For this one specifically I believe the remote API is `RegisterPool`, not `RequestPool`.